app/services/dttp/access_token_service.rb

```diff
@@ -0,0 +1,47 @@
+module Dttp
+  class AccessTokenService
+    class AuthenticationClient
+      include HTTParty
+
+      MICROSOFT_LOGIN_URL = "https://login.microsoftonline.com".freeze
+
+      base_uri MICROSOFT_LOGIN_URL
+    end
+
+    def self.call
+      new.call
+    end
+
+    def call
+      options = {
+        body: {
+          grant_type: "client_credentials",
+          client_id: client_id,
+          scope: scope,
+          client_secret: client_secret,
+        },
+      }
+
+      response = AuthenticationClient.post("/#{tenant_id}/oauth2/v2.0/token", options)
+      JSON.parse(response.body)["access_token"]
```

Comment:
do we need to do any sanity checks for or log for failures?

Reply:
I don't think we can recover from it really so it's probably sufficient to let the error bubble out to sentry and let sidekiq retry. We can have a rethink of that as we build the layers up as per the comment with David on this PR.